src/js/pagination/paging-collection.js

```diff
@@ -83,7 +83,15 @@
              * backbone.paginator.
              */
             parseState: function (response, queryParams, state, options) {
-                return {totalRecords: response[0].count, totalPages: response[0].num_pages};
+                state.totalRecords = response[0].count;
+                state.totalPages = response[0].num_pages;
+
+                // Remove already processed response
+                delete response.count;
+                delete response.num_pages;
+
+                // Update the state with response
+                return PageableCollection.prototype.parseState.call(this, response, queryParams, state, options);
             },
```

Comment:
While doing a reset or set on collection directly the state doesn't set with server returned or passed data. So calling parent's (PageableCollection) parseState ensures that state is in sync.